pkg/bootkube/bootkube.go

```diff
@@ -86,8 +87,23 @@ func (b *bootkube) Run() error {
 	}
 
 	if selfHostedEtcd {
+		useEtcdTLS, err := detectEtcdTLS(b.assetDir)
+		if err != nil {
+			return err
+		}
+		var tlsConfig *tls.Config
```

Comment:
Well, I meant you could move all of this to etcdutil too. Maybe just pass the assetDir into Migrate() and you can construct the other two paths inside that method as well.